### DISCUSSION TITLE
Is it possible to connect to Doris with Rust?

### DISCUSSION THREAD

**Discussion by tumluliu** [apache/doris#40507] [A - General / Q&A]
there is no rust sample code from the `samples/connect`. Am I supposed to use the sqlx-mysql to connect to it? Thanks!

**Comment by ixzc** [comment]
you can use mysql to connect. doris is highly compatible with MySQL

**Reply by tumluliu** [reply]
Thank you for your reply. Yes, I finally managed to communicate with Doris and saw the SQL results with sqlx. 

But for other Rust devs who may be interested, it's required to explicitly call `.no_engine_subsitution(false).pipes_as_concat(false)` when initializing the `MySqlConnectOptions`

**Reply by eyeori** [reply]
sql statements  like `SHOW PROC '/tasks/PUBLISH_VERSION';` cannot be executed successfully using sqlx, as they return an `Unsupported command (COMSTMT_PREPARE)` error.

**Comment by zhangruchubaba** [comment]
@tumluliu Have you implemented using Rust to access Starrocks
